src/components/Sidebar/PostDetails/NewCommentForm.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { addComment } from '../../../utils/fetchFunctions';
+import {
+  ActivePostContext,
+  CommentListContext,
+  ErrorsContext,
+} from '../../../utils/Store';
+import { ErrorTypes } from '../../../types/ErrorTypes';
+
+enum Keys {
+  nameField = 'nameField',
+  emailField = 'emailField',
+  textArea = 'textArea',
+}
+
+export const NewCommentForm: React.FC = () => {
+  const [isErrorField, setIsErrorField] = useState({
+    nameField: false,
+    emailField: false,
+    textArea: false,
+  });
+
+  const [inputQueries, setInputQueries] = useState({
+    nameField: '',
+    emailField: '',
+    textArea: '',
+  });
+
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const { activePost } = useContext(ActivePostContext);
+  const { comments, setComments } = useContext(CommentListContext);
+  const { isError, setIsError } = useContext(ErrorsContext);
+
+  const onChangeHandler = (property: Keys, value: string) => {
+    if (isErrorField[property]) {
+      setIsErrorField({
+        ...isErrorField,
+        [property]: false,
+      });
+    }
+
+    setInputQueries({ ...inputQueries, [property]: value });
+  };
+
+  function handleClear() {
+    setInputQueries({
+      nameField: '',
+      emailField: '',
+      textArea: '',
+    });
+
+    setIsErrorField({
+      nameField: false,
+      emailField: false,
+      textArea: false,
+    });
+  }
+
+  const handleAdd = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newError = { ...isErrorField };
+
+    Object.values(Keys).forEach(key => {
+      if (!inputQueries[key].trim()) {
+        newError[key] = true;
+      }
+    });
+
+    setIsErrorField(newError);
+
+    const error = Object.values(newError).some(v => v);
+
+    if (!error) {
+      setIsProcessing(true);
+
+      const newComment = {
+        id: 0,
+        postId: activePost?.id as number,
+        name: inputQueries.nameField,
+        email: inputQueries.emailField,
+        body: inputQueries.textArea,
+      };
+
+      addComment(newComment)
+        .then(res => {
+          if (isError) {
+            setIsError(null);
+          }
+
+          setComments([...comments, res]);
+          setInputQueries({ ...inputQueries, textArea: '' });
+        })
+        .catch(() => {
+          setIsError(ErrorTypes.Comment);
+        })
+        .finally(() => {
+          setIsProcessing(false);
+        });
+    }
+  };
```

Comment:
The function `handleAdd` uses `setState` multiple times within the same function call. Consider combining these `setState` calls into a single call to improve code clarity and performance.